File: tests/benchmark_scope.py
import time

def currentTimeMillis():
    return int(time.time() * 1000)

def benchmark():
    a = "global"
    # keep the same “nested scopes” vibe using nested functions
    def scope1():
        b = "first"
        def scope2():
            c = "second"
            def scope3():
                d = "third"
                def scope4():
                    e = "fourth"
                    def scope5():
                        f = "fifth"

                        start = currentTimeMillis()
                        i = 0
                        while i < 1_000_000:
                            x = a
                            y = b
                            z = e
                            i += 1
                        end = currentTimeMillis()
                        print("Time taken:", (end - start), "ms")
                    scope5()
                scope4()
            scope3()
        scope2()
    scope1()

benchmark()

File: tests/benchmark_scope.cpp
#include <chrono>
#include <iostream>
#include <string>

long currentTimeMillis() {
  using namespace std::chrono;
  return duration_cast<milliseconds>(system_clock::now().time_since_epoch())
      .count();
}

int main() {
  std::string a = "global";
  {
    std::string b = "first";
    {
      std::string c = "second";
      {
        std::string d = "third";
        {
          std::string e = "fourth";
          {
            std::string f = "fifth";
            // Access variables from different depths 1,000,000 times
            long start = currentTimeMillis();
            int i = 0;
            while (i < 1000000) {
              std::string x = a;
              std::string y = b;
              std::string z = e;
              i = i + 1;
            }
            long end = currentTimeMillis();
            std::cout << "Time taken: " << (end - start) << "ms" << std::endl;
          }
        }
      }
    }
  }
  return 0;
}
